src/store/singing.ts

```diff
@@ -99,22 +116,67 @@ const generateAudioEvents = async (
   return [{ time, buffer }];
 };
 
-const generateNoteEvents = (notes: Note[], tempos: Tempo[], tpqn: number) => {
-  return notes.map((value): NoteEvent => {
-    const noteOnPos = value.position;
-    const noteOffPos = value.position + value.duration;
-    return {
-      noteNumber: value.noteNumber,
-      noteOnTime: tickToSecond(noteOnPos, tempos, tpqn),
-      noteOffTime: tickToSecond(noteOffPos, tempos, tpqn),
+const setupAudioEvents = async (
+  audioContext: BaseAudioContext,
+  channelStrip: ChannelStrip,
+  transport: Transport | OfflineTransport,
+  singingGuides: Map<SingingGuideSourceHash, SingingGuide>,
+  singingVoices: Map<SingingVoiceSourceHash, SingingVoice>,
+  phrases: Phrase[],
+) => {
+  for (const phrase of phrases) {
+    if (
+      phrase.singingGuideKey == undefined ||
+      phrase.singingVoiceKey == undefined ||
+      phrase.state !== "PLAYABLE"
+    ) {
+      continue;
+    }
+    const singingGuide = singingGuides.get(phrase.singingGuideKey);
+    const singingVoice = singingVoices.get(phrase.singingVoiceKey);
+    if (!singingGuide) {
+      throw new Error("singingGuide is undefined");
+    }
+    if (!singingVoice) {
+      throw new Error("singingVoice is undefined");
+    }
+    // TODO: この辺りの処理を共通化する
+    const audioEvents = await generateAudioEvents(
+      audioContext,
+      singingGuide.startTime,
+      singingVoice.blob,
+    );
+    const audioPlayer = new AudioPlayer(audioContext);
+    const audioSequence: AudioSequence = {
+      type: "audio",
+      audioPlayer,
+      audioEvents,
     };
-  });
+    audioPlayer.output.connect(channelStrip.input);
+    transport.addSequence(audioSequence);
+  }
 };
 
 let audioContext: AudioContext | undefined;
 let transport: Transport | undefined;
 let previewSynth: PolySynth | undefined;
-let channelStrip: ChannelStrip | undefined;
+let previewChannelStrip: ChannelStrip | undefined;
+let globalChannelStrip: ChannelStrip | undefined;
+
+// Undoしたときに不明なトラックを参照してしまうので、DefaultMapを使う。
+// TODO: 通常のMapに変更する
+const channelStrips = new DefaultMap<TrackId, ChannelStrip>(() => {
```

Comment:
「一元管理されてる状態変数のみが状態を持っているべき」的な法則からすると、trackをwatchして変更時にこっちも変更する形が良さそうに感じました。
`SingEditor.vue`内でcomposable作るとかすればできそうですが、いろいろ大変だと思うので、とりあえずちょっとTODOを補強するのはどうでしょう。
```suggestion
// Undoしたときに不明なトラックを参照してしまうので、DefaultMapを使う。
// TODO: 通常のMapにし、state.tracksをwatchして逐次変更する
const channelStrips = new DefaultMap<TrackId, ChannelStrip>(() => {
```